molo/core/models.py

```diff
@@ -626,10 +626,14 @@ def save(self, *args, **kwargs):
         if (languages.filter(
                 is_main_language=True).exists() and
                 not self.languages.exists()):
+            language = languages.filter(
+                is_main_language=True).first()
             LanguageRelation.objects.create(
                 page=self,
-                language=languages.filter(
-                    is_main_language=True).first())
+                language=language)
+            response = super(
+                TranslatablePageMixinNotRoutable, self).save(*args, **kwargs)
```

Comment:
I don't understand this line or why it's required since it is already present at the start of this method. I don't believe it is in any way related to adding the fields so I don't think it should be in this PR.